### ISSUE TITLE
Paralell programming

### ISSUE THREAD

**Issue by balazsszekely** [Microsoft/ClearScript#12]
Hi!

Is there any way to run multiple tasks at the the same time from javascript code without create new V8ScriptEngine instance? The tasks would use javascript code.

For example:

`var task = System.Task.Run(new System.Action(function() { System.Console.WriteLine("I'm in the new task."); }));
task.Wait();
`

Thanks,
Balazs

**Comment by NKumarPraveen** [comment]
You can run parallel script using java script promise objects.
[https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise)

**Comment by ClearScriptLib** [comment]
@balazsszekely If you're using a single instance of `V8ScriptEngine`, you can execute JavaScript code on any thread, but not on multiple threads simultaneously.

Your example launches a JavaScript task on a worker thread, but waits for it on the current thread. This is a deadlock because the worker thread can't start running JavaScript code until the current thread stops doing so.

ClearScript supports asynchronous callbacks into script code, so as long as your scripts don't do any waiting or perform other long-running operations, you should be able to take full advantage of .NET's threading capabilities.

JavaScript's Promise objects don't provide concurrency by themselves, but they provide a standard API pattern that you can use to wrap the concurrency capabilities of the host.

**Comment by balazsszekely** [comment]
Thanks for fast reply!
